platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -210,71 +199,82 @@ public Factory saveFactory(Iterator<FileItem> formData,
     }
 
     /**
-     * Save factory to storage.
+     * Stores the factory from the configuration.
      *
      * @param factory
      *         instance of factory which would be stored
-     * @return decorated factory instance of which has been stored
-     * @throws org.eclipse.che.api.core.ApiException
-     *         @link org.eclipse.che.api.core.BadRequestException} when factory is null
-     *         @link org.eclipse.che.api.core.ConflictException} when problem occurs during factory storing
-     *         @link org.eclipse.che.api.core.ServerException} when internal server error occurs
+     * @return decorated the factory instance of which has been stored
+     * @throws BadRequestException
+     *         when stored the factory is invalid
+     * @throws ServerException
+     *         when any server errors occurs
+     * @throws ForbiddenException
+     *         when the user have no access rights for saving the factory
+     * @throws ConflictException
+     *         when stored the factory is already exist
      */
-    @RolesAllowed("user")
     @POST
-    @Consumes({MediaType.APPLICATION_JSON})
-    @Produces({MediaType.APPLICATION_JSON})
-    public Factory saveFactory(Factory factory) throws ApiException {
+    @Consumes({APPLICATION_JSON})
+    @Produces({APPLICATION_JSON})
+    @RolesAllowed("user")
+    @ApiOperation(value = "Stores the factory from the configuration",
+                  notes = "Stores the factory without pictures and returns instance of the stored factory with links",
+                  response = Factory.class)
```

Comment:
response is not required for methods which explicitly define their response class